backend/api/communication/src/main/java/co/airy/core/api/communication/AsyncSendMessagesHandler.java

```diff
@@ -0,0 +1,159 @@
+package co.airy.core.api.communication;
+
+import co.airy.avro.communication.Channel;
+import co.airy.avro.communication.ChannelConnectionState;
+import co.airy.avro.communication.DeliveryState;
+import co.airy.avro.communication.Message;
+import co.airy.avro.communication.Metadata;
+import co.airy.log.AiryLoggerFactory;
+import co.airy.model.conversation.Conversation;
+import co.airy.model.metadata.MetadataKeys;
+
+import java.time.Duration;
+import java.time.Instant;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutionException;
+
+import javax.annotation.PostConstruct;
+import javax.annotation.PreDestroy;
+import javax.validation.constraints.Min;
+
+import org.apache.kafka.streams.state.ReadOnlyKeyValueStore;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.stereotype.Component;
+import org.slf4j.Logger;
+
+import lombok.AllArgsConstructor;
+import lombok.Data;
+
+import static co.airy.model.metadata.MetadataRepository.newMessageMetadata;
+
+@Component
+public class AsyncSendMessagesHandler implements Runnable {
+
+    private static final Logger log = AiryLoggerFactory.getLogger(AsyncSendMessagesHandler.class);
+
+    @AllArgsConstructor
+    @Data
+    private class PendingMessage {
+        private Message msg;
+        private Instant now;
+    }
+
+    private final Stores stores;
+    private final ConcurrentLinkedQueue<PendingMessage> pendingConversations;
+    private final Thread thread;
+    private final long maxWaitMillis;
+    private final long periodMillis;
+    private volatile boolean keepAlive;
+    
+
+    AsyncSendMessagesHandler(
+            Stores stores,
+            @Value("${async-messages-handler.max-wait-millis}")
+            long maxWaitSeconds,
+            @Min(1)
+            @Value("${async-messages-handler.check-period-millis}")
+            long periodMillis) {
+        this.stores = stores;
+        this.pendingConversations = new ConcurrentLinkedQueue<>();
+        this.keepAlive = true;
+        this.maxWaitMillis = maxWaitSeconds;
+        this.periodMillis = periodMillis;
+        this.thread = new Thread(this);
+    }
+
+    @Override
+    public void run() {
+        final List<PendingMessage> pending = new LinkedList<>();
+
+        while (keepAlive) {
+            try {
+                Thread.sleep(this.periodMillis);
```

Comment:
I think an easier way to do this would be to use [`@Scheduled`](https://github.com/airyhq/airy/blob/0f3ab3554fbc41b0fe1f44d04fb32c858a1c84b3/backend/api/admin/src/main/java/co/airy/core/api/config/ServiceDiscovery.java#L53)

But in general I wonder if there is a way that lets us avoid maintaining this linked list as it's prone to fail. How about having something like
```java
@Async 
public void sendMessage(Message message) {
}

@Scheduled() // with some config
private void trySendMessage(Message message) {
// ...
}
```
that relies on the async executor service we use for PNS and the [`@Retry`](https://www.baeldung.com/spring-retry) annotation.

More info: https://docs.spring.io/spring-batch/docs/current/reference/html/retry.html

What do you think?